spec/inputs/rabbitmq_spec.rb

```diff
@@ -166,21 +166,73 @@ def spawn_and_wait(instance)
   end
 
   describe "receiving a message with a queue specified" do
-    let(:queue_name) { "foo_queue" }
+    let(:queue_name) { "logstash-input-rabbitmq-#{rand(0xFFFFFFFF)}" }
     let(:config) { super.merge("queue" => queue_name) }
+    let(:queue) { test_channel.queue(queue_name, :auto_delete => true) }
 
     it "should process the message" do
       message = "Foo Message"
-      q = test_channel.queue(queue_name)
-      q.publish(message)
+      queue.publish(message)
 
       event = output_queue.pop
       expect(event["message"]).to eql(message)
     end
+
+    it "should save message properties into a @metadata field" do
+      epoch = 1454789198
+      # Don't test every single property but select a few with
+      # different characteristics to get sufficient coverage.
+      queue.publish("",
+                    :properties => {
+                      :app_id    => "myapplication",
+                      :timestamp => Java::JavaUtil::Date.new(epoch * 1000),
+                      :priority  => 5,
+                    })
+
+      event = output_queue.pop
+      expect(event).to include("@metadata")
+      expect(event["@metadata"]).to include("rabbitmq_properties")
+
+      props = event["@metadata"]["rabbitmq_properties"]
+      expect(props["app-id"]).to eq("myapplication")
+      expect(props["delivery-mode"]).to eq(1)
+      expect(props["exchange"]).to eq("")
+      expect(props["priority"]).to eq(5)
+      expect(props["routing-key"]).to eq(queue_name)
+      expect(props["timestamp"]).to eq(epoch)
+    end
+
+    it "should save message headers into a @metadata field" do
+      headers = {
+        "arrayvalue"  => [true, 123, "foo"],
+        "boolvalue"   => true,
+        "intvalue"    => 123,
+        "stringvalue" => "foo",
+      }
+      queue.publish("",
+                    :properties => {
+                      :headers => headers,
+                    })
+
+      event = output_queue.pop
+      expect(event).to include("@metadata")
+      expect(event["@metadata"]).to include("rabbitmq_headers")
+      expect(event["@metadata"]["rabbitmq_headers"]).to include(headers)
+    end
+
+    it "should have an empty message header hash if message has no headers" do
+      queue.publish("")
+
+      event = output_queue.pop
+      expect(event).to include("@metadata")
+      expect(event["@metadata"]).to include("rabbitmq_headers")
+      expect(event["@metadata"]["rabbitmq_headers"]).to eq({})
+    end
   end
 
   describe LogStash::Inputs::RabbitMQ do
-    let(:config) { super.merge("queue" => "foo_queue") }
+    # With auto_delete this example just hangs
```

Comment:
@magnusbaeck you can enable logs by setting the env var `TEST_DEBUG=true` . Mind trying that out?